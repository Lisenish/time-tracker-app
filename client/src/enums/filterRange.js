export const FilterRange = Object.freeze({
  DAY: "day",
  WEEK: "week",
  MONTH: "month"
});
