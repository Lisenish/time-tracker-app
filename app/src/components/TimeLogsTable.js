import FormControl from "@material-ui/core/FormControl";
import InputLabel from "@material-ui/core/InputLabel";
import MenuItem from "@material-ui/core/MenuItem";
import Paper from "@material-ui/core/Paper";
import Select from "@material-ui/core/Select";
import { makeStyles } from "@material-ui/core/styles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import React from "react";
import { FilterRange } from "../enums/filterRange";
import formatMsToTimeString from "../services/time-formatter";

const useStyle = makeStyles({
  filter: {
    minWidth: "120px",
    marginLeft: "auto"
  }
});
export default function TimeLogsTable({ timeLogs, selectedFilter, onFilterChange }) {
  const classes = useStyle();

  return (
    <TableContainer component={Paper}>
      <Toolbar>
        <Typography variant="h5">Time sessions</Typography>

        <FormControl className={classes.filter}>
          <InputLabel id="filter-select-label">Filter</InputLabel>
          <Select
            id="filter-select"
            labelId="filter-select-label"
            value={selectedFilter}
            onChange={onFilterChange}
          >
            <MenuItem value={FilterRange.DAY}>Day</MenuItem>
            <MenuItem value={FilterRange.WEEK}>Week</MenuItem>
            <MenuItem value={FilterRange.MONTH}>Month</MenuItem>
          </Select>
        </FormControl>
      </Toolbar>
      <Table aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>Session name</TableCell>
            <TableCell align="right">Time</TableCell>
            <TableCell align="right">Date</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {timeLogs.map(log => (
            <TableRow key={log.id}>
              <TableCell component="th" scope="row">
                {log.name}
              </TableCell>
              <TableCell align="right">
                {formatMsToTimeString(log.time)}
              </TableCell>
              <TableCell align="right">
                {log.createdAt.toLocaleDateString()}
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}
