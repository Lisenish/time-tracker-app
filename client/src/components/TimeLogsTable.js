import FormControl from "@material-ui/core/FormControl";
import InputLabel from "@material-ui/core/InputLabel";
import MenuItem from "@material-ui/core/MenuItem";
import Paper from "@material-ui/core/Paper";
import Select from "@material-ui/core/Select";
import { makeStyles } from "@material-ui/core/styles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import React from "react";
import { FilterRange } from "../enums/filterRange";
import formatMsToTimeString from "../services/time-formatter";

const useStyle = makeStyles({
  filter: {
    minWidth: "120px",
    marginLeft: "auto"
  }
});

export default function TimeLogsTable({
  timeLogs,
  selectedFilter = FilterRange.DAY,
  onFilterChange
}) {
  const classes = useStyle();

  return (
    <TableContainer component={Paper}>
      <TableToolbar
        className={classes.filter}
        selectedFilter={selectedFilter}
        onFilterChange={onFilterChange}
      />
      <Table aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>Session name</TableCell>
            <TableCell align="right">Time</TableCell>
            <TableCell align="right">Date</TableCell>
          </TableRow>
        </TableHead>

        <TableBody>
          {timeLogs && timeLogs.length ? (
            timeLogs.map(createTimeLogRow)
          ) : (
            <EmptyState />
          )}
        </TableBody>
      </Table>
    </TableContainer>
  );
}

function TableToolbar({ className, selectedFilter, onFilterChange }) {
  return (
    <Toolbar>
      <Typography variant="h5">Time sessions</Typography>

      <FormControl className={className}>
        <InputLabel id="filter-select-label">Filter</InputLabel>
        <Select
          id="filter-select"
          labelId="filter-select-label"
          value={selectedFilter}
          onChange={onFilterChange}
        >
          <MenuItem value={FilterRange.DAY}>Day</MenuItem>
          <MenuItem value={FilterRange.WEEK}>Week</MenuItem>
          <MenuItem value={FilterRange.MONTH}>Month</MenuItem>
        </Select>
      </FormControl>
    </Toolbar>
  );
}

function EmptyState() {
  return (
    <TableRow>
      <TableCell colSpan={3} align="center">
        No recorded sessions
      </TableCell>
    </TableRow>
  );
}

function createTimeLogRow(timeLog) {
  return (
    <TableRow key={timeLog.id}>
      <TableCell component="th" scope="row">
        {timeLog.name || "Unnamed session"}
      </TableCell>
      <TableCell align="right">{formatMsToTimeString(timeLog.time)}</TableCell>
      <TableCell align="right">
        {timeLog.createdAt.toLocaleDateString()}
      </TableCell>
    </TableRow>
  );
}
