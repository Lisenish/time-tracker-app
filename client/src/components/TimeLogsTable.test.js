import React from "react";
import renderer from "react-test-renderer";
import TimeLogsTable from "./TimeLogsTable";
import { FilterRange } from "../enums/filterRange";

const testLogs = [
  { id: 1, name: "Test Session", time: 1000, createdAt: new Date("01-01-2020") }
];

test("Table renders correctly with zero state", () => {
  const tree = renderer.create(<TimeLogsTable />).toJSON();

  expect(tree).toMatchSnapshot();
});

test("Table renders correctly with empty logs array", () => {
  const tree = renderer.create(<TimeLogsTable testLogs={[]} />).toJSON();

  expect(tree).toMatchSnapshot();
});

test("Table renders correctly with logs", () => {
  const tree = renderer
    .create(
      <TimeLogsTable timeLogs={testLogs} selectedFilter={FilterRange.DAY} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test("Table renders correctly selected filter WEEK", () => {
  const tree = renderer
    .create(
      <TimeLogsTable timeLogs={testLogs} selectedFilter={FilterRange.WEEK} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
