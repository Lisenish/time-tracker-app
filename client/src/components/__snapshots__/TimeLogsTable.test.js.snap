// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly selected filter WEEK 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      Time sessions
    </h5>
    <div
      className="MuiFormControl-root makeStyles-filter-1"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
        data-shrink={true}
        id="filter-select-label"
      >
        Filter
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="filter-select-label filter-select"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          id="filter-select"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Week
        </div>
        <input
          onAnimationStart={[Function]}
          type="hidden"
          value="week"
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
  <table
    aria-label="simple table"
    className="MuiTable-root"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Session name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Time
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Date
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    >
      <tr
        className="MuiTableRow-root"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          scope="row"
        >
          Test Session
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          00:00:01
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          1/1/2020
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table renders correctly with empty logs array 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      Time sessions
    </h5>
    <div
      className="MuiFormControl-root makeStyles-filter-1"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
        data-shrink={true}
        id="filter-select-label"
      >
        Filter
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="filter-select-label filter-select"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          id="filter-select"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Day
        </div>
        <input
          onAnimationStart={[Function]}
          type="hidden"
          value="day"
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
  <table
    aria-label="simple table"
    className="MuiTable-root"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Session name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Time
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Date
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    >
      <tr
        className="MuiTableRow-root"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
          colSpan={3}
        >
          No recorded sessions
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table renders correctly with logs 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      Time sessions
    </h5>
    <div
      className="MuiFormControl-root makeStyles-filter-1"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
        data-shrink={true}
        id="filter-select-label"
      >
        Filter
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="filter-select-label filter-select"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          id="filter-select"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Day
        </div>
        <input
          onAnimationStart={[Function]}
          type="hidden"
          value="day"
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
  <table
    aria-label="simple table"
    className="MuiTable-root"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Session name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Time
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Date
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    >
      <tr
        className="MuiTableRow-root"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          scope="row"
        >
          Test Session
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          00:00:01
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          1/1/2020
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table renders correctly with zero state 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      Time sessions
    </h5>
    <div
      className="MuiFormControl-root makeStyles-filter-1"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
        data-shrink={true}
        id="filter-select-label"
      >
        Filter
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="filter-select-label filter-select"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          id="filter-select"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Day
        </div>
        <input
          onAnimationStart={[Function]}
          type="hidden"
          value="day"
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
  <table
    aria-label="simple table"
    className="MuiTable-root"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Session name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Time
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Date
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    >
      <tr
        className="MuiTableRow-root"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
          colSpan={3}
        >
          No recorded sessions
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
